src/analyzer/analyzerbeats.cpp

```diff
@@ -229,57 +228,58 @@ void AnalyzerBeats::storeResults(TrackPointer tio) {
         QHash<QString, QString> extraVersionInfo = getExtraVersionInfo(
                 m_pluginId, m_bPreferencesFastAnalysis);
         pBeats = BeatFactory::makePreferredBeats(
-                *tio,
+                pTrack,
                 beats,
                 extraVersionInfo,
                 m_bPreferencesFixedTempo,
                 m_bPreferencesOffsetCorrection,
-                m_iSampleRate,
                 m_iTotalSamples,
                 m_iMinBpm,
                 m_iMaxBpm);
         qDebug() << "AnalyzerBeats plugin detected" << beats.size()
-                 << "beats. Average BPM:" << (pBeats ? pBeats->getBpm() : 0.0);
+                 << "beats. Average BPM:" << (pBeats ? pBeats->getBpm().getValue() : 0.0);
     } else {
-        float bpm = m_pPlugin->getBpm();
+        mixxx::Bpm bpm = mixxx::Bpm(m_pPlugin->getBpm());
```

Comment:
We use static casts in this case, to have it clear distinguished from a function call.